app/javascript/src/NewService/components/NewServiceForm.jsx

```diff
@@ -0,0 +1,145 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+import {ServiceSourceForm, FormTemplate} from 'NewService'
+import type {SourceFormProps, FormTemplateProps} from 'NewService/types'
+
+const BASE_URL = '/p/admin/service_discovery/'
+const NAMESPACES_URL = `${BASE_URL}projects.json`
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+const fetchData = async (url: string) => {
+  const dataItem = url === NAMESPACES_URL
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url).then(response => {
+    if (!response.ok) {
+      throw Error(response.statusText)
+    }
+    return response.json()
+  })
+    .then(data => data[dataItem].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => console.error(error))
+}
+
+const getSourceFormProps = (props, handleFormsVisibility, startNamespacesFetch) => {
+  return {
+    isServiceDiscoveryUsable: props.isServiceDiscoveryUsable,
+    serviceDiscoveryAuthenticateUrl: props.serviceDiscoveryAuthenticateUrl,
+    onHandleFormsVisibility: handleFormsVisibility,
+    onStartNamespacesFetch: startNamespacesFetch
+  }
+}
+
+const getServiceDiscoveryProps = (props, isServDescVisible, namespaces, services, selectedNamespace, onChangeNamespaces) => {
+  return {
+    isFormDiscovery: true,
+    isVisible: isServDescVisible,
+    formActionPath: props.providerAdminServiceDiscoveryServicesPath,
+    namespaces: namespaces,
+    services: services,
+    selectedNamespace: selectedNamespace,
+    handleChangeNamespaces: onChangeNamespaces
+  }
+}
+
+const getServiceNewProps = (props, isServNewVisible) => {
+  return {
+    isFormDiscovery: false,
+    isVisible: isServNewVisible,
+    formActionPath: props.adminServicesPath
+  }
+}
+
+const callEfect = (servicesRequested, selectedNamespace, startServicesFetch) => {
+  if (servicesRequested && selectedNamespace) {
+    const SERVICES_URL = `${BASE_URL}namespaces/${selectedNamespace}/services.json`
+    startServicesFetch(SERVICES_URL)
+  }
+}
+
+const ErrorMessage = () => <p style={{color: 'red'}}>Sorry, your request has failed!</p>
```

Comment:
@guicassolato 
At the moment, when fetch fails we only have a console log, but no error to the user in the UI.
I'm adding this basic error message, what do you think?
![error-message](https://user-images.githubusercontent.com/13486237/57436350-c9767300-723f-11e9-9dcd-5e442b154d35.png)

Reply:
I think it's a good idea though we need a better error message.

Reply:
Do we know why it has failed @damianpm?